hometask7/tests/game.test.js

```diff
@@ -0,0 +1,119 @@
+import { Game } from "../src/game";
+import { Player } from "../src/player";
+
+describe("The test environment", () => {
+  const game = new Game([new Player("p1"), new Player("p2")]);
+  const fakeFn = () => {};
+  test("addPlayer()", () => {
+    game.addPlayer(new Player("p3"));
+    expect(game.players.players[2].name).toBe("p3");
+  });
+  test("run()", () => {
+    const spy = jest.spyOn(game, "roll").mockImplementation(fakeFn);
+    game.run();
+    expect(spy).toHaveBeenCalled();
+  });
+  test("roll() rollGenerator", () => {
+    const game1 = new Game([new Player("p1"), new Player("p2")]);
+    const spyRollGenerator = jest
```

Comment:
TODO: spies look a little bit messy. You can create separate function that will create a spy on class function and mock implementation.

Something like:

```
function spyAndMock(obj, method, implementation) {
 return jest
      .spyOn(obj, method)
      .mockImplementation(implementation);
}
```